### PR TITLE
Feature/mismatched webservice version

### PR DESCRIPTION
**Pull request by agduncan94**
On staging the banner will now specify if the webservice the UI is talking to is the expected one.

https://github.com/ga4gh/dockstore/issues/1308

### CODE REVIEW

File: src/app/banner/banner.component.ts [hunk 1]
```diff
@@ -9,10 +13,32 @@ import { environment } from '../../environments/environment';
...
+  mismatchedVersion = false;
+  versionFromAPI: string;
+  versionBuiltWith: string;
+
+  constructor(private gA4GHService: GA4GHService) { }
 
   ngOnInit() {
     this.showBanner = environment.staging;
+
+    this.gA4GHService.metadataGet()
```
Comment (garyluu) [inline src/app/banner/banner.component.ts]: this looks like it does a 2nd metadataGet call when the footer already did one.  double check if there's 1 or 2 metadata get calls every time the application is loaded.  If it's duplicated, somehow make the footer and banner use metadata from the same call.

**Comment by agduncan94** [comment]
@garyluu Made changes you mentioned

File: src/app/metadata/metadata.service.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
+import { Metadata } from './../shared/swagger/model/metadata';
+import { HttpErrorResponse } from '@angular/common/http';
+import { map } from 'rxjs/operators';
+import 'rxjs/add/operator/publishReplay';
+
+@Injectable()
+export class MetadataService {
+  metadata: any;
```
Comment (garyluu) [inline src/app/metadata/metadata.service.ts]: Missing type here as well as methods

File: src/app/banner/banner.component.ts [hunk 3]
```diff
@@ -9,10 +12,32 @@ import { environment } from '../../environments/environment';
...
+  versionFromAPI: string;
+  versionBuiltWith: string;
+
+  constructor(private metadataService: MetadataService) { }
 
   ngOnInit() {
     this.showBanner = environment.staging;
+
+    this.metadataService.getMetadata()
+      .subscribe(
```
Comment (garyluu) [inline src/app/banner/banner.component.ts]: Missing unsubscribe, add unsubscribe to footer while ur at it.

File: src/app/banner/banner.component.ts [hunk 4]
```diff
@@ -9,10 +12,32 @@ import { environment } from '../../environments/environment';
...
+        (metadata: Metadata) => {
+          if (metadata.hasOwnProperty('version')) {
+            this.versionFromAPI = metadata['version'];
+            this.versionBuiltWith = versions.version;
```
Comment (garyluu) [inline src/app/banner/banner.component.ts]: Can probably move this out of unsubscribe

Comment (agduncan94) [reply src/app/banner/banner.component.ts]: Do you mean out of subscribe? And move it to a function?

Comment (garyluu) [reply src/app/banner/banner.component.ts]: out of subscribe

**Pull request merged**